src/eponai/tools/cursive_arcadia_repl.clj

```diff
@@ -52,6 +60,18 @@
               (assoc :id id))))))))
 
 
+(defn arcadia-nrepl-eval [handler msg]
+  (try
+    (arcadia-handle-message handler msg)
+    (catch SocketException e
+      (print "Arcadia connection lost, reconnecting...")
+      (connect!)
```

Comment:
How about surrounding the retry logic around this `connect!` function and add a sleep?

Reply:
What do you mean we should surround it by?

Reply:
Talked in person, and decided this is out of scope. This PR is only fixing the problem with the REPL disconnecting when the game runs, and we don't need the retry logic for that.